base_layer/core/src/validation/block_validators.rs

```diff
@@ -118,17 +119,23 @@ fn check_coinbase_output(block: &Block) -> Result<(), ValidationError> {
     block.check_coinbase_output().map_err(ValidationError::from)
 }
 
+// This function checks that all inputs in the blocks are valid UTXO's to be spend
 fn check_inputs_are_utxos<B: BlockchainBackend>(
-    _block: &Block,
-    _db: BlockchainDatabase<B>,
+    block: &Block,
+    db: BlockchainDatabase<B>,
 ) -> Result<(), ValidationError>
 {
-    // TODO --implement Issue #1092
+    for utxo in block.body.inputs() {
+        if !(utxo.features.flags.contains(OutputFlags::COINBASE_OUTPUT)) &&
+            !(db.is_utxo(utxo.hash())).map_err(|_| ValidationError::NoDatabaseConfigured)?
```

Comment:
`is_utxo` will throw a ChainStorageError, which does not mean that db is not set (this should only throw if `db.is_none()`

```suggestion
            !(db.is_utxo(utxo.hash())).map_err(|e| ValidationError::CustomError(e.to_string())?
```